connectors/yarpc/yarpc.go

```diff
@@ -633,8 +634,11 @@ func (c *Connector) CreateScope(ctx context.Context, scope string) error {
 
 // TruncateScope truncates all data in the scope specified
 func (c *Connector) TruncateScope(ctx context.Context, scope string) error {
+	// A better authn story is needed -- an evildoer could just craft a request with a
+	// bogus owner name and send it directly, bypassing this client.
 	request := &dosarpc.TruncateScopeRequest{
-		Name: &scope,
+		Name:      &scope,
+		Requester: dosa.GetUsername(),
```

Comment:
how do you plan to authenticate this? Does the the requester of `TruncateScope` needs to be the same as the requester of `CreateScope`?  I suppose there could be multiple users accessing the same scope.

Reply:
Same comment for all three methods -- right now it's easy to spoof, but after goldstar phase 2 (or whatever it's called) we should be able to do it right, with user-level authn/authz.

Reply:
... I guess we will have to figure out some sort of pluggable module thing for authz/authn, I don't goldstar is open-sourced.

Reply:
Yes I think it makes sense that only the owner of a scope can drop or truncate it.

Reply:
A scope should be owned by a team, not a particular user imo. Another option is to allow them specify the owner in the CLI, but that's also easy to spoof and we'll need to think about the migration of existing scopes. 

Is there a doc that I should look at :)

Reply:
Hmmm team owned scopes are an interesting point. The only reasonable way to do it is to use LDAP groups, and confirm that the requesting user is in the requested group... and the server should be the one to verify that.

I'm going to add another field to the IDL, `owner`.  The user can use any string, the client won't verify it; the server will confirm that the owner is either the same as requester, or an LDAP group that the requester is part of.


Doc: https://docs.google.com/document/d/1sTehpUFV8qcbx_AlsXPmwlWyse3PvgnGcF59OEFPI4U/edit#

Reply:
Updated the doc with team scopes

Reply:
Looks good to me